### DISCUSSION TITLE
Questions about the new "Battery 1" entity

### DISCUSSION THREAD

**Discussion by amit-nz** [wlcrs/huawei_solar#742] [General]
My setup contains a SUN2000 inverter, two power meters (I have another non-huawei inverter), one DC-DC converter and 2 x 5 kWh modules and a huawei backup box. I run it in "Maximize self consumption" mode always.

Just upgraded to the latest version of the integration. I see that it now exposes "Battery 1" - it looks like this this is my battery stack (which includes 1 x DC-DC converter, and two x 5 kWh modules). 

Super happy to be able to keep an eye on temps now! I have a couple of questions:

Firstly - What is the main difference between the "Battery" and "Battery 1" as shown under the integration? Looks like a lot of data is duplicated (with the exception of a couple of entities like Temperature).

Secondly - I see that there is a "Battery 1 Remaining charge/discharge time" entity. I presumed this would show me ETA under force charge/discharge from/to grid or solar charge conditions, but it always remains at "0 minutes". Is this broken or am I not understanding exactly how it works?

**Comment by LotharWoman** [comment]
I think "Battery 1 Remaining charge/discharge time" is just named incorrectly. This sensor is certainly intended for BackupTime. So the time in which the battery supplied the BackUpBox in the event of a power failure.

**Comment by Roving-Ronin** [comment]
Hi @amit-nz 

The new 'Batteries' sensors provide the readings from the modbus registers that reflect the overall total (for that particular reading/sensor) for all batteries attached to the inverter (or inverters in a cascade).

The Battery 1 / Battery 2 etc sensors provide the details for the individual battery stacks.

Note there is a difference between what is being reported for Batteries and a individual Battery (as noticed with a single LUNA S0 installed). 
See: https://github.com/wlcrs/huawei_solar/issues/720  I have flagged this with the Australian OEM to see if they can raise this to Huawei to fix.

____________________

@LotharWoman 

The "Battery 1 Remaining charge/discharge time" isn't incorrectly named, it is an actual modbus register.

![image](https://github.com/wlcrs/huawei_solar/assets/108674933/601544d2-3183-4e5e-854f-aa9b7079789f)

However I would agree its a 'weird' sensor/register, as 'Energy Storage Unit 1' in Huawei documents refers to the 'Battery 1' (as this integration calls it). Looking through the rest of the 'Huawei Modbus Register Definitions' documentation, there is NO register for 'Energy Storage Unit 2' !  This is strange and you'd think either there would be a register for each of the 2 batteries that can attached to each inverter (and the 4 in future with the LUNA S1 7/13/21kWh model) or if there is only one register, that it would be for 'Energy Storage Remaining charge/discharge time' (i.e. for the Batteries device, that shows the sum data for all installed batteries).

In regards the issue your seeing, with the sensor remaining at '0 min', I can confirm I am also seeing this issue after 2-3 days since updating (and then deleting and reinstalling the plant). See issue logged at: https://github.com/wlcrs/huawei_solar/issues/741 . I would suggest you also jump on there and comment that you have this issue.
